service/keyv2/key.go

```diff
@@ -45,6 +47,19 @@ func CrtTTL(customObject v1alpha1.CertConfig) string {
 	return customObject.Spec.Cert.TTL
 }
 
+func CustomObjectHash(customObject v1alpha1.CertConfig) (string, error) {
+	b, err := json.Marshal(customObject.Spec.Cert)
```

Comment:
So the idea is to use the config hash to detect changes.

Reply:
I wouldn't say this is deterministic. Do you have any guarantees on field order inside json?
If there is any map inside I'm pretty sure the output is randomized on purpose.

Reply:
There are only booleans strings and string slices: https://github.com/giantswarm/apiextensions/blob/f0b8bcd6122deaf8be38b9705ac528cf4b749e3a/pkg/apis/core/v1alpha1/cert_types.go#L61-L70.

Reply:
Are you sure json fiels will be always marshaled in the same order? What happens when they aren't?

Reply:
The JSON implementation is deterministic with the current types. It is always alphabetical. In case this breaks, certs are renewed more often than they should. We should improve on metrics and see when we renew too often. That would be cool. There is nothing that can horribly break out of a sudden though IMO. 

Please feel free to suggest a better approach here if necessary. I understand your concerns. I was thinking about several ways to do it and this was the best I was able to come up with in the scope of the postmortem issue.

Reply:
> The JSON implementation is deterministic with the current types. It is always alphabetical.

When that is true this is fine. I just wanted to make sure you are aware of that.

Also if renewing certs is not a problem, then I'm 100% happy.

Reply:
Thanks for the discussion.